File: tests/workflow/producer/rust/src/producer/events/groupb_structa.rs
use super::{identification, producer::Control, protocol, Context};
use crate::test::samples;
use clibri::server;
use uuid::Uuid;

type BroadcastStructA = (Vec<Uuid>, protocol::StructA);
type BroadcastStructB = (Vec<Uuid>, protocol::StructB);

#[allow(unused_variables)]
pub async fn emit<E: server::Error, C: server::Control<E> + Send + Clone>(
    event: protocol::GroupB::StructA,
    filter: &identification::Filter,
    context: &mut Context,
    control: &Control<E, C>,
) -> Result<(BroadcastStructA, BroadcastStructB), String> {
    Ok((
        (filter.all(), samples::struct_a::get()),
        (filter.all(), samples::struct_b::get()),
    ))
}

File: experimental/workflow_mt/producer/typescript/src/producer/events/groupb.structa.ts
import { Filter, Producer, Context, Protocol } from "../implementation/events";
import { Output } from "../implementation/events/groupb.structa";

export function emit(
    event: Protocol.GroupB.StructA,
    filter: Filter,
    context: Context,
    producer: Producer
): Promise<Output> {
    return Promise.reject(
        new Error(`Handler for event "GroupB.StructA" isn't implemented`)
    );
}